### PR TITLE
BO - Import files - Do not show generated csv files from import/csvfromexcel folder in history of uploaded files

### LINKED ISSUE #22747: [EPIC] - Import / Export
All issues related to BO>Import /Export

### Bugs
- [x] #9951
- [x] #9995
- [x] #10874
- [ ] #12867
- [x] #22236
- [ ] #21565
- [ ] #25215
- [ ] #26157
- [ ] #21565
- [ ] #27900 
- [ ] #29432
- [x] #32688
- [x] #34601 
- [ ] #34966

**Customers**
- [x] #23106
- [ ] #32658

**Products**
- [x] #9677
- [x] #9771
- [ ] #10824
- [x] #12092
- [x] #13509
- [ ] #14934
- [ ] #21682
- [x] #29254

**Combinations**
- [x] #17604
- [x] #23083

### Features
- [x] #11497

### Bugs
- [x] #15334
- [x] #28228
- [x] #22372
- [x] #28761
- [x] #32413

**Customers**
- [x] #24583

**Products**
- [x] #22336
- [x] #14306

**Combinations**
- [x] #12321

### Features
- [x] #24802
- [x] #28073

### LINKED ISSUE #28761: BO - An exception is thrown when import a file with non .csv extension and download the csv created file
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

An exception is thrown when import a file with non .csv extension and download the csv created file 

image
 1786 and 1778 

Uploading exception when download csv file .mp4…

### Expected behavior

Only the uploaded files by the user are listed in the 'Choose from history / FTP'. 
The files generated from the import system are not listed in the 'Choose from history / FTP' and can't be downloadable.

### Steps to reproduce

1. Go to CONFIGURE > Advanced Parameters > Import
2. Choose a **non .csv** (eg.odt, xsl, xlst, etc) file by clicking in select a file to import 
3. Click on Next step
4. Click on Cancel
5. See in Choose from history / FTP that a csv file is created
6. Click to download it 
7. See error > Exception

### PrestaShop version(s) where the bug happened

1.7.7.8 & develop & 1.7.8.6

### PHP version(s) where the bug happened

7.4 / 7.2

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by mflasquin**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Only the uploaded files by the user are listed in the 'Choose from history / FTP'. Exclude files subdirectories and especially from import/csvfromexcel.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28761 
| Related PRs       | 
| How to test?      | See #28761
| Possible impacts? |

### CODE REVIEW

**Review by matks — DISMISSED** [review]

**Review by matks — COMMENTED** [review]
Nice-to-have: do you think you could have a look at this class and see if we could test it?

We have other Test cases that do move files in `tests/Unit`

The directory https://github.com/PrestaShop/PrestaShop/tree/develop/tests/Unit/Resources contains sample files that can be used in such files

**Review by yanmakouf — DISMISSED** [review]
Seems ok to me. Just wondering if it would be difficult to make those generated files store in a different folder than the one uploaded by the user. So we don't need to rely on depth(). Just a guess ^^

File: tests/Unit/Core/Import/File/FileFinderTest.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    /** @var ImportDirectory */
+    protected $importDirectory;
+
+    /** @var Filesystem  */
+    protected $fs;
+
+    protected function setUp(): void
+    {
+        $configuration = new ConfigurationMock();
```
Comment (atomiix) [inline tests/Unit/Core/Import/File/FileFinderTest.php]: Usually we use `$this->createMock()` and redefine methods that needs to be mocked but I guess it works with an override 🤷‍♂️

Comment (mflasquin) [reply tests/Unit/Core/Import/File/FileFinderTest.php]: You're right, it's better. It's fixed :)

**Review by matks — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by matthieu-rolland — COMMENTED** [review]
Maybe we should put this test in `Test/Integration/Core/Import...` since you are dealing the system (file system) without mocking it, I'd consider it an integration test

but other opinions are welcome

**Review by matthieu-rolland — DISMISSED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by 0x346e3730 — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by HanaRebaiQA — APPROVED** [review]
Hello @mflasquin 

I have checked your PR. The issue is well fixed.

Check the automated tests: OK
Check with different and valid file extensions (ods, odt, xlsx, xlst, xls csv) : OK
Check with different and invalid file extensions (pdf, apk, mp4, mp3, rar, jpg,...) : OK
Check with multistore: OK

https://user-images.githubusercontent.com/92912932/212642061-d56ad386-47b2-4b08-b466-78d6b24bcf33.mp4


So, It's QA :heavy_check_mark: 

Thank You!

**Comment by nicosomb** [comment]
Thank you @mflasquin !

**Pull request merged**